src/index.js

```diff
@@ -242,9 +242,12 @@ export function renderer(view, state, actions) {
         out += frame.footer
         stack.pop()
       } else {
-        const node = resolveNode(frame.children[frame.childIndex++], state, actions)
+        let node = resolveNode(frame.children[frame.childIndex++], state, actions)
 
         if (node != null && typeof node !== 'boolean') {
+          if (node.type === 2 && node.lazy) {
```

Comment:
Are you sure we need both checks, can't we leave only one?

Reply:
I was thinking about it myself and decided to take a defensive approach here. I couldn't think of a scenerio where only one of the 2 would be present though. If we're to leave only one do you prefer the node.type === 2? You already have node.type === 3 check in the code.